### PR TITLE
KW41Z 802.15.4 driver: Enable OpenThread support 

### PR DESCRIPTION
**Pull request by dleach02**
Added support to enable OpenThread over KW41Z driver.

I'm able to see the OT network come up and both sides communicating and establishing the Parent. But many times I see problems with the echo client/server where one side will decide that the partner is not one that can be sent to even though it is receiving packets from it. Not sure yet if I'm missing something on the higher level configuration of the OT.

Apologies for the duplicate PR. I closed the other one to fix some problems and now it won't let me reopen it.

### CODE REVIEW

File: drivers/entropy/Kconfig.mcux [hunk 1]
```diff
@@ -19,7 +19,7 @@ menuconfig ENTROPY_MCUX_TRNG
 	depends on ENTROPY_GENERATOR && HAS_MCUX_TRNG
 	default n
 	select ENTROPY_HAS_DRIVER
-	select ENTROPY_DEVICE_RANDOM_GENERATOR
+	select XOROSHIRO_RANDOM_GENERATOR
```
Comment (rlubos) [inline drivers/entropy/Kconfig.mcux]: Just wondering... Is this anyhow related to OpenThread port?

File: drivers/flash/soc_flash_mcux.c [hunk 2]
```diff
@@ -89,7 +89,7 @@ static int flash_mcux_write(struct device *dev, off_t offset,
 	addr = offset + priv->config.PFlashBlockBase;
 
 	key = irq_lock();
-	rc = FLASH_Program(&priv->config, addr, (u32_t *) data, len);
+	rc = FLASH_Program(&priv->config, addr, (uint32_t *) data, len);
```
Comment (rlubos) [inline drivers/flash/soc_flash_mcux.c]: Any particular reason `uint32_t` is used here?

**Review by rlubos — COMMENTED** [review]
Good to see that more platforms are supporting OpenThread in Zephyr. Just FYI, I've verified this PR on nRF52840 locally, no issues detected.

File: drivers/flash/soc_flash_mcux.c (hunk 2 above)
Comment (dleach02) [reply drivers/flash/soc_flash_mcux.c]: Because for some reason the KW41Z XCVR code causes the use of NewLib. NewLib uses a different typedef for u32_t that is not portable to a uint32_t which results in the gcc compiler producing a warning. I put some of the details in the check-in notes. GCC will warn that an 'unsigned int' is not generally equal to 'uint32_t' because the size of an 'unsigned int' is an architecture specific item while C ensures that a uint32_t is always 32 bits.

File: drivers/ieee802154/ieee802154_kw41z.c [hunk 3]
```diff
@@ -24,6 +24,61 @@
...
+		u16_t frame_version		:2;
+		u16_t dst_addr_mode		:2;
+		u16_t ie_list			:1;
+		u16_t seq_num_suppr		:1;
+#endif
+	} fc __packed;
+
+	u8_t sequence;
+} __packed;
+#endif
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_kw41z.c]: Let me fix this the issue #5942 so you won't need this.

**Review by tbursztyka — DISMISSED** [review]
All looks fine to me, I am just locking it the time I fix #5942 quickly

File: drivers/ieee802154/ieee802154_kw41z.c (hunk 3 above)
Comment (dleach02) [reply drivers/ieee802154/ieee802154_kw41z.c]: I think the real fix for this is to just put the 802.15.4 packet framing definitions into the common directory tree. I know that in my previous PR you didn't want to do this because you felt this was internal information to the 15.4 native stack but I don't look at it that way. Even Linux has this information in a common include tree.

Comment (tbursztyka) [reply drivers/ieee802154/ieee802154_kw41z.c]: I know Linux does it like that, but I really prefer to keep the crude details of framing into its current place. Also, if for some reason one wants to rewrite the framing logic or even replace the L2 by another one, it will be easier that way.

Comment (dleach02) [reply drivers/ieee802154/ieee802154_kw41z.c]: We are talking about the framing details of an IEEE standard. That is why I would suggest having an ieee802154.h in the common include space that only has typedefs and stuff for the 15.4 protocol as defined in the standard.

Comment (tbursztyka) [reply drivers/ieee802154/ieee802154_kw41z.c]: The way frames are parsed here is all but standard. You could do it in different ways.

Comment (dleach02) [reply drivers/ieee802154/ieee802154_kw41z.c]: You control this part but your point about this being easier if we replace L2 isn't clear to me. All I'm talking about is the 802.15.4 protocol, which these drivers are providing HW access for. You are already adding this special part to the OT enablement to add in part of the code that is part of the native stack. The drivers should be agnostic to what is attaching to it and not need any additional special hooks to get to this information. My view is that the native 15.4 stack, OT, and any other protocol stack we may put on top should be as self contained as possible and we should avoid borrowing pieces of the stacks to accomplish what is really just an IEEE802.15.4 protocol detail and not tied to OT or our native 15.4 stack..

Comment (tbursztyka) [reply drivers/ieee802154/ieee802154_kw41z.c]: I see, let me update #6631 I think I have a better compromise that will suit our both opinions: no frame headers being public at well as no specific part of the L2 being built for that.

**Comment by dleach02** [comment]
I did rebase with the new ar flag changes...

**Comment by tbursztyka** [comment]
@dleach02 tree is changing quite a lot sometimes, once rebased I'll ask @jukkar to apply it. It seems fine to me.

**Review by tbursztyka — APPROVED** [review]

**Review by jukkar — APPROVED** [review]
LGTM

**Review by jfischer-no — COMMENTED** [review]
LGTM

**Comment by dleach02** [comment]
Does anything else need to occur for this to merge?
